File: JavaScript/8js 2/isOpposite.js
/*
|--------------------------------------------------------------------------
| Are they opposite?
|--------------------------------------------------------------------------
|
*/

function isOpposite(s1, s2) {
   return (
      s1
         .split("")
         .map((i) => (i === i.toLowerCase() ? i.toUpperCase() : i.toLowerCase()))
         .join("") === s2 && s1 !== ""
   );
}


function isOpposite(s1, s2) {
   return (
      s1 != "" &&
      s1 == s2.replace(/./g, (ch) => ch["to" + (ch <= "Z" ? "Lower" : "Upper") + "Case"]())
   );
}


function isOpposite(s1, s2) {
   return (s1 == "" && s2 == "") || s1.length != s2.length
      ? false
      : [...s1].every((x, i) => Math.abs(x.charCodeAt() - s2[i].charCodeAt()) == 32);
}


console.log(isOpposite("ab", "AB"), true);
console.log(isOpposite("aB", "Ab"), true);
console.log(isOpposite("aBcd", "AbCD"), true);
console.log(isOpposite("aBcde", "AbCD"), false);
console.log(isOpposite("AB", "Ab"), false);
console.log(isOpposite("", ""), false);

File: Python/8py 2/is_opposite.py
"""
|--------------------------------------------------------------------------
| Are they opposite?
|--------------------------------------------------------------------------
|
"""

def is_opposite(s1,s2):
   return "".join([i.upper() if i.islower() else i.lower() for i in s1]) == s2 and s1 != ""


def is_opposite(s1, s2):
   return bool(s1) and s1.swapcase() == s2


def is_opposite(s1,s2):
   return s1 == s2.swapcase() if s1 else False


print(is_opposite("ab","AB") , True);
print(is_opposite("aB","Ab") , True);
print(is_opposite("aBcd","AbCD") , True);
print(is_opposite("AB","Ab") , False);
print(is_opposite("","") , False);